integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -364,7 +364,12 @@ object DataLoadProcessBuilderOnSpark {
     TaskContext.get.addTaskCompletionListener { _ =>
       CommonUtil.clearUnsafeMemory(ThreadLocalTaskInfo.getCarbonTaskInfo.getTaskId)
     }
-    TaskMetricsMap.threadLocal.set(Thread.currentThread().getId)
+    if (TaskMetricsMap.getThreadLocal.get() == null) {
```

Comment:
Move this complete if condition to `TaskMetricsMap` and expose method to initialize threadlocal

Reply:
done